### PR TITLE
celery control cleanup

### PR DESCRIPTION
**Pull request by biblicabeebli**
(This is the second in a sequence of pull requests containing several bugfixes and rebases)

This is a refactoring and separation of different components of celery, firebase, and push notification code.

Rebase pull request of the first pull request:
https://github.com/onnela-lab/beiwe-backend/pull/231

original session timestamp bugfix pull request:
https://github.com/onnela-lab/beiwe-backend/pull/221

There have been some posted errors similar to this, Alvin posted a fix that hit most cases at some point in the past month, but I just spun up a local server from scratch and immediately hit this bug after logging in as the default admin.

Fix is to test the correct date-time object in the ~authentication decorator.

### CODE REVIEW

**Review by zagorsky — CHANGES_REQUESTED** [review]
@biblicabeebli I think you forgot to update all instances of `libs.push_notification_config`.  Here's what I get when I run `python app.py`:

```
Traceback (most recent call last):
  File "app.py", line 19, in <module>
    from pages import (admin_pages, data_access_web_form, forest_pages, login_pages, mobile_pages,
  File "/home/jzag88/beiwe-backend/pages/participant_pages.py", line 12, in <module>
    from libs.push_notification_config import (check_firebase_instance,
ModuleNotFoundError: No module named 'libs.push_notification_config'
```

Please fix and update this PR, and re-request review after you've tested running the server locally and using it to send push notifications to phones via ngrok.

**Comment by biblicabeebli** [comment]
Ack, this is resolved over in commit 23ff75221e79ffa57739a889b0394a9e4a9d3ab9, which is ahead another few commits past the 3rd branch in the sequence pull requests I've already created.  This is because I merged in the elements I intended to have on the final tested code on the eli/sender-mismatch-exceptions branch.  The code was not broken when written back on Aug 15.

@zagorsky: what process do you want here?  If you want I can go solve this with some fiddling, (Does a cherry-pick and then a rebase of the original code would poke the bits around correctly?)

(aaaand it looks like the previous pull request approval modified the commit message... with a co-author statement? changing the commit hash between main and the former session-timestamp-bugfix branch's commit.  So that's a thing.

File: libs/firebase_config.py
Comment (hackdna) [inline libs/firebase_config.py:39]: Since prefix `test_` is commonly used for test suite functions, rename to something like `check_firebase_credential_errors`?

**Comment by zagorsky** [comment]
@biblicabeebli sorry if rebases caused your code to break.  Also, as you observed: GitHub's PR merge process means that the head commit on a branch changes (the commit hash changes) when the PR gets merged via the GitHub web interface, which means that despite all the work you put in making this PR be a ff-merge ahead of the previous PR, once the previous PR got merged, Git no longer sees it that way.  Although the diff itself will still be clean, the history looks like it diverged.

But I don't want to merge this PR until it's working.  So I think the best course of action is for you to rebase this PR onto `main` (which shouldn't mean any material code changes; just changes in commit hashes), resolve any bugs via cherry-pick, manual edits, whatever you like, and then re-request review on it.

And in the future, if you have multiple PRs open, if they don't interact with each other, it's probably fine to make them all branch directly off of `main` instead of branching off of each other.

**Comment by biblicabeebli** [comment]
(Aaaaand typed this up but forgot to hit comment 12 hours ago.)

@zagorsky done with git magic to make it maximally clean and work.

Rebased on top of the update to main.
cherry-picked the commit with the fix that was several ahead
rebasing the diverged branches on top of the new eli/celer_control-cleanup elided the "original" fix commit into a null diff, and has disappeared it.

If sir would continue his review...

**Review by zagorsky — APPROVED** [review]
Code looks good, and I also just tested this locally and am still able to send push notifications to a phone (registered via ngrok).

@biblicabeebli, you should decide if you want to follow @hackdna's [comment](https://github.com/onnela-lab/beiwe-backend/pull/232#discussion_r712480448) and rename `test_firebase_credential_errors()` to `check_firebase_credential_errors()`, and either rename the function or not.

After that, I think you can go ahead and merge this into `main`- I'd recommend using the "Squash and merge" button in GitHub.

**Comment by biblicabeebli** [comment]
This code should not be squashed, it is already a combination of two merged branches accomplishing different things for cleanup.  @zagorsky This branch is strictly a ff-merge, does the rebase and merge option fully preserve the current commit hash (unlike merge pull request which can mutate the commit message?) so that I don't have to then go and re-rebase the next two branches in order to keep the network graph clean?

@hackdna: [haha what tests.]
Fair point - I've _absolutely_ encountered dumb scoping issues involving the word "test" in filenames and Test classes due to the aggressiveness of Nose / the DjangoTestRunner searching for it specifically, but I've not hit an issue like that in an external namespace.  I really want this code out already and the existing rebase crud has already delayed it.  I'm going to throw a note onto my todo list to check the system for functions starting with test in all scopes over on my eli/django branch, as the bulk of future tests will run through django.

**Comment by zagorsky** [comment]
@biblicabeebli the GitHub "Rebase and Merge" button will change the commit hashes of all the commits on this branch ([see this StackOverflow question](https://stackoverflow.com/questions/48350294/)).  If you're desperate to maintain the commit hashes so you can avoid rebasing the other PRs, then you should merge this one into main via the command line and push it.

**Comment by zagorsky** [comment]
@biblicabeebli I merged it into `main` for you, since it looks like that's what you wanted to do.

**Pull request merged**
